packages/handlersjs-http/lib/services/sync.service.spec.ts

```diff
@@ -0,0 +1,288 @@
+
+  // saves the latest "modified since" header from mocked requests
+  let latestModifiedSince: string | undefined;
+
+  beforeEach(() => {
+
+    jest.resetAllMocks();
+
+    fetchMock = (fetch as jest.MockedFunction<typeof fetch>);
+    latestModifiedSince = undefined;
+
+    fetchMock.mockImplementation(async (url, options) => {
+
+      latestModifiedSince = options?.headers['If-Modified-Since'];
+
+      return new Response('[]', { status: 200, headers });
+
+    });
+
+    store = new MemoryStore<M>([
+      [ 'storage', new Set<number>() ],
+      [ 'peers', new Set<string>(peers) ],
+      [ 'other', 'something else' ],
+    ]);
+
+    syncService = new SyncService<number, 'storage', 'peers', M>('storage', 'peers', store);
+
+  });
+
+  interface peerResponseMock {
+    peer: string;
+    httpStatus: number;
+    storages?: number[];
+  }
+
+  // mocks the fetch() function with http responses according to the peerResponseMock interface
+  const mockWithStorages = (... responseMocks: peerResponseMock[]) => {
+
+    fetchMock.mockImplementation(async (url, options) => {
+
+      latestModifiedSince = options?.headers['If-Modified-Since'];
+
+      for (const mockInfo of responseMocks) {
+
+        if (mockInfo.peer === url) {
+
+          return new Response(JSON.stringify(mockInfo.storages ? mockInfo.storages : []),
+            { status: mockInfo.httpStatus, headers });
+
+        }
+
+      }
+
+      return new Response('[]', { status: 200, headers });
+
+    });
+
+  };
+
+  describe('sync()', () => {
+
+    it('resolves when there are no peers', async () => {
+
+      await store.set('peers', new Set());
+      await expect(syncService.sync()).resolves.toBe(syncService);
+
+    });
+
+    it('resolves when peers are undefined', async () => {
+
+      await store.delete('peers');
+      await expect(syncService.sync()).resolves.toBe(syncService);
+
+    });
+
+    it('sends a request to all peers in the store', async () => {
+
+      await syncService.sync();
+      fetchMock.mock.calls.forEach((call) => expect(peers.has(call[0].toString())).toBe(true));
+
+    });
+
+    describe('If-Modified-Since behavior', () => {
+
+      it('doesn\'t have the If-Modified-Since header on the first request', async () => {
+
+        await syncService.sync();
+        expect(latestModifiedSince).toBeUndefined();
+
+      });
+
+      it('has the If-Modified-Since header from the second request and onwards', async () => {
+
+        await syncService.sync();
+        expect(latestModifiedSince).toBeUndefined();
+        await syncService.sync();
+        expect(latestModifiedSince).toBeDefined();
+        await syncService.sync();
+        expect(latestModifiedSince).toBeDefined();
+
+      });
+
+      it('has the correct modifiedSince header', async () => {
+
+        let beforeFirstSync = Date.now();
+        beforeFirstSync -= beforeFirstSync % 1000; // If-Modified-Since header value rounds down to the second
```

Comment:
This probably isn't the most clean method to get time in seconds.

Reply:
Can only find solutions online with division by 1000, which we would need to multiple again anyways, so I'm not sure

Reply:
Why would we need to multiply it again? What is wrong with the following?
```suggestion
        const beforeFirstSync = Math.floor(Date.now() / 1000); // If-Modified-Since header value rounds down to the second
```

Reply:
i didn't do this because it would have to be three times instead of once then (for all the other values that will be compared), but I implemented it with division now